### PR TITLE
Build magma for windows

### LINKED ISSUE #139397: Build a non-conda package for `magma-cuda`
As a first step, this can be done using the conda base image, so all dependencies are already in place.
Next we need to switch to a non-conda base image.

cc @malfet @seemethere

### LINKED ISSUE #140001: Implement the Windows builds as well

### PR DESCRIPTION
**Pull request by afrittoli**
Copy the magma for windows job and script from pytorch/builder https://github.com/pytorch/builder/blob/c9aac65e12734a16d9c285c366bcbf6cc0c67e43/.github/workflows/build-magma-windows.yml

The linux version is moved here in https://github.com/pytorch/pytorch/pull/139888

Fixes #140001

### CODE REVIEW

File: .github/workflows/build-magma-windows.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Push MAGMA to S3
+        run: |
+          # Setting the following variable avoids errors in GHA https://github.com/aws/aws-cli/issues/5623
+          cd artifact
+          export AWS_EC2_METADATA_DISABLED=true
+          set -x
+          ${UPLOAD_COMMAND} . $OSSCI_WINDOWS_S3 --exclude="*" --include="magma_*_cuda*.7z" --recursive --acl public-read
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_OSSCI_S3_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_OSSCI_S3_ACCESS_KEY }}
```
Comment (afrittoli) [inline .github/workflows/build-magma-windows.yml:66]: This requires the corresponding secrets being defined in pytorch/pytorch

**Comment by afrittoli** [comment]
Will update this PR once https://github.com/pytorch/pytorch/pull/139888 is merged, to use the same upload script.

File: .github/workflows/build-magma-windows.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    branches:
+      main
+    paths:
+      - .github/scripts/windows/*
+      - .github/workflows/build-magma-windows.yml
+  pull_request:
+    paths:
+      - .github/scripts/windows/*
+      - .github/workflows/build-magma-windows.yml
+
```
Comment (atalman) [inline .github/workflows/build-magma-windows.yml:14]:
```suggestion
    permissions:
      id-token: write
      contents: read
```

Comment (afrittoli) [reply .github/workflows/build-magma-windows.yml:14]: The `id-token` permission is only needed by the `push` job, so I added it on L48 (see below)

**Review by atalman — APPROVED** [review]
lgtm

**Pull request closed without merging**